### PR TITLE
Add increment and decrement functionality to left resource quantity 

### PR DESCRIPTION
**Pull request by BelieveC**
Addresses issue #1284 . Improve resource show page with an addition of increment and decrement functionality for Available quantity and Total left quantity using Ajax. 
increment

### CODE REVIEW

**Comment by siddhantbajaj** [comment]
Great @BelieveC . There are some indentation issues in the changes you have made. You can have a look at them  here https://travis-ci.org/openSUSE/osem/builds/202780655 or by looking the details of your test.

**Comment by BelieveC** [comment]
@siddhantbajaj, Fixed indentation issues.

File: app/controllers/admin/resources_controller.rb [hunk 1]
```diff
@@ -39,7 +39,39 @@ def destroy
...
+
+    def increment_quantity
+      @resource.quantity = @resource.quantity + 1
+      @resource.save
+      respond_to do |format|
+        format.js{ render action: 'update_quantity' }
+      end
+    end
+
+    def decrement_quantity
```
Comment (Ana06) [inline app/controllers/admin/resources_controller.rb]: `increment_quantity`, `decrement_quantity` are exactly the same method except by the fact that one increment the variable and the other one decrements it. to avoid repeating code what about having only `increment_quantity` with a parameter, something like:


``` Ruby
  def increment_quantity(step = 1)
      @resource.quantity = @resource.quantity + step
      @resource.save
      respond_to do |format|
        format.js{ render action: 'update_quantity' }
      end
    end
```

and then

`increment_quantity` and `increment_quantity(-1)`

Moreover this would allow to change the increment step easily in case that is needed.

File: app/controllers/admin/resources_controller.rb [hunk 2]
```diff
@@ -39,7 +39,39 @@ def destroy
       else
         redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
                     error: 'Resource was successfully destroyed.' \
-  	                "#{@resource.errors.full_messages.join('. ')}."
+                    "#{@resource.errors.full_messages.join('. ')}."
```
Comment (Ana06) [inline app/controllers/admin/resources_controller.rb]: why did you change this line? :confused:

**Comment by differentreality** [comment]
The buttons are way too tiny. My finger is bigger than both of them, how can this be used with touch screens?

Also can't we use update to change the used attribute? Instead of introducing 2 new actions?

**Comment by BelieveC** [comment]
@differentreality, We can do it using update action but I think it would be better to keep them as separate action because we can improve increment and decrement functionality with giving user option to increment or decrement it by more than 1 step if required. i.e Decrement by 10 or 100 and so on. What do you think @differentreality ?

**Comment by differentreality** [comment]
But this is not what we are currently doing, nor do I see a use case for decreasing more than 1 per time.

**Comment by BelieveC** [comment]
Sure, I would make requested changes.

**Comment by BelieveC** [comment]
@differentreality, I have removed 2 new actions. Incrementing/Decrementing it using update action.

File: app/controllers/admin/resources_controller.rb (hunk 2 above)
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Sure! I will remove it.

File: app/controllers/admin/resources_controller.rb [hunk 3]
```diff
@@ -23,12 +23,19 @@ def create
...
-      else
-        flash[:error] = "Resource update failed: #{@resource.errors.full_messages.join('. ')}."
-        render :edit
+      respond_to do |format|
+        if @resource.update_attributes(resource_params)
+          format.html do
+            redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
+                        notice: 'Resource successfully updated.'
+          end
+          format.js{ render action: 'update_quantity' }
```
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Adding an error handler for the else part.

Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Added flash message for else part
error

**Comment by BelieveC** [comment]
- Index page view in Desktop resolution 
dindex

- Show page view in Desktop resolution 
dshow

- Index page view in Mobile resolution 
mindex

- Show page view in Mobile resolution
mshow

**Comment by BelieveC** [comment]
@differentreality, How about this

1. 0c0080808579d9addc5c233718731bd1476e23d6

2.jquery-plugin-for-increment-decrement-number-spinner-stepperwidget

**Comment by differentreality** [comment]
screenshot from 2017-02-28 11-53-35

Your +/- buttons are not aligned. Perhaps we should have another column just for the buttons? I think that would make them look nicer, what do you think?

This is showing that I have 153 tshirts left out of 150 tshirts that I had in total! That shouldn't be possible, both validations-wise and UI-wise, the button shouldn't be enabled unless I can click on it, otherwise the user will click whatever he or she wants. 

Used column does not change when I click on the +/- buttons.

File: app/views/admin/resources/_update_actions.html.haml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+= link_to admin_conference_resource_path(conference.short_title, resource,resource:{ used: resource.used-1 }), method: :put, remote: true do
+	%span.fa.fa-plus-circle.fa-2x.update_action
+
+= link_to admin_conference_resource_path(conference.short_title, resource,resource:{ used: resource.used + 1 }), method: :put, remote: true do
+	%span.fa.fa-minus-circle.fa-2x.update_action
```
Comment (differentreality) [inline app/views/admin/resources/_update_actions.html.haml]: Empty line at the end of your file!

File: app/views/admin/resources/index.html.haml [hunk 5]
```diff
@@ -16,14 +16,16 @@
...
-              %td
+              %td.col_vertical_align
                 = link_to(admin_conference_resource_path(@conference.short_title, resource.id)) do
                   = resource.name
                 %td
-                  = quantity_left_of(resource)
-                %td
-                  = resource.used
-                %td
+                  %span{ class: "quantity_left", id: "resource_left_#{resource.id}" }= quantity_left_of(resource)
```
Comment (differentreality) [inline app/views/admin/resources/index.html.haml]: Please use single quotes unless there is string interpolation.

Comment (differentreality) [inline app/views/admin/resources/index.html.haml]: Move the code inside your %span, not next to it.

File: app/views/admin/resources/_update_actions.html.haml (hunk 4 above)
Comment (BelieveC) [inline app/views/admin/resources/_update_actions.html.haml]: Sorry! Will fix it.

**Comment by BelieveC** [comment]
@differentreality, I don't think adding a new column for increment/decrement button would look that good even though it would be properly aligned. I think it would be ambiguous to which column it tend to increment and decrement. Below are two different design, need some suggestion on them.

1. 
d1

2.
d2

**Comment by differentreality** [comment]
Second design also looks nice, but please commit changes so that we can review it.

**Comment by BelieveC** [comment]
Thanks, Will commit changes with the second design!

**Comment by BelieveC** [comment]
@differentreality, Following are screenshot index and show page in mobile and desktop resolution

1. Desktop (Show page) 
dindex

2. Desktop (Index page) 
dshow

3. Mobile (Index page) 
mindex

4. Mobile(Show page) 
mshow

**Comment by BelieveC** [comment]
@differentreality, Please can you review the changes?

**Comment by differentreality** [comment]
For mobile devices, doesn't it make sense to have the two buttons further away? For example, minus button left or top, plus button right or bottom.

I am not sure we need this layout for the show page though.

File: app/views/admin/resources/_update_actions.html.haml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+.btn-group
+	%button.btn.btn-default.quantity_left.disabled
+		%span{ id: "resource_left_#{resource.id}" }
+			=quantity_left_of(resource)
+	%button{ class: "btn btn-default #{ 'disabled' if resource.used == resource.quantity }" }
+		=link_to admin_conference_resource_path(conference.short_title, resource,resource:{ used: resource.used + 1 }), method: :put, remote: true do
+			%span.fa.fa-minus.fa-2x.update_action
+	%button{ class: "btn btn-default #{ 'disabled' if resource.used == 0  }" }
+		=link_to admin_conference_resource_path(conference.short_title, resource,resource:{ used: resource.used-1 }), method: :put, remote: true do
+			%span.fa.fa-plus.fa-2x.update_action
```
Comment (differentreality) [inline app/views/admin/resources/_update_actions.html.haml]: Please add empty line at the end of the file, it will make this red indication go away.

**Comment by differentreality** [comment]
Can we really not live without adding all that new CSS?

File: app/views/admin/resources/index.html.haml [hunk 7]
```diff
@@ -16,14 +16,16 @@
...
                 = link_to(admin_conference_resource_path(@conference.short_title, resource.id)) do
                   = resource.name
                 %td
-                  = quantity_left_of(resource)
-                %td
-                  = resource.used
-                %td
+                  %span{ id: "update_actions_#{resource.id}" }
+                    = render partial: 'update_actions', locals: { conference: @conference, resource: resource }
+                %td.vertical_align
```
Comment (differentreality) [inline app/views/admin/resources/index.html.haml]: Align this column right, it will look better and it will be easier to read the numbers

**Comment by differentreality** [comment]
What about showing some flash messages on quantity change so that we know what we just did?

File: app/views/admin/resources/_update_actions.html.haml (hunk 6 above)
Comment (BelieveC) [inline app/views/admin/resources/_update_actions.html.haml]: Thanks! Will add it.

File: app/views/admin/resources/index.html.haml (hunk 7 above)
Comment (BelieveC) [inline app/views/admin/resources/index.html.haml]: :+1:

**Comment by BelieveC** [comment]
I will try my best to minimize used CSS. I am not sure about flash message similar to notification flash we use. We may try something else like toast notification or Some animation on increased or decreased quantity e.g Twitter like button. If not flash message would be a default way to do it. What do you think?

**Comment by BelieveC** [comment]
For mobile devices, Design with Minus at top and Plus at the bottom makes column width too large for index page table. Keeping them on left and right respectively within possible width for a mobile resolution put a constraint on the size of a button and makes size too small. We try different layout for show page. Will get you some design :+1: Still will be happy to know your suggestion on a show page. :+1:

**Comment by differentreality** [comment]
Let's stick to our regular flash messages. There are many things we can do, and I am sure everyone has a different idea on how to implement it, that doesn't mean we should do them all. Besides changing how we show messages is out of the scope of this PR. Use unobtrusive flash which we already have to show the messages please like the rest of the app does.

Goal should be for organizers to be able to use their mobile devices (touch screen) without pushing - when they want to push +, which is why I am suggesting we put those buttons as far away as possible.
Looking forward to your screenshots.

**Comment by BelieveC** [comment]
I will make necessary changes :+1:

**Comment by BelieveC** [comment]
@differentreality,
1. Desktop (Show page) 
dshow

2. Desktop (Index page) 
dindex

3. Mobile (Index page) 
mindex


4. Mobile(Show page) 
mshow

Looking forward for suggestion.

**Comment by differentreality** [comment]
1. Do we truly need the extra CSS you have added in this PR?
2. The logic for the controller method to reduce resources is wrong. Try the following scenario: 
deviceA reduces resource from 5 to 4, deviceB (that had loaded the page before deviceA reduced the resource) reduces the same resource. Expected left amount of recources is 3, but we get 4.
3. I am open to out-of-the-box suggestions for a nicer index (?) page that would make this feature more functional to its true use (which is not just to see a list of our available resources - that's just a small part of it), sadly I am not very good with front-end ideas.

**Comment by BelieveC** [comment]
@differentreality 
> Do we truly need the extra CSS you have added in this PR?

Yes, We require that CSS. I have tried to minimize the use of external CSS. 

> The logic for the controller method to reduce resources is wrong. Try the following scenario:
deviceA reduces resource from 5 to 4, deviceB (that had loaded the page before deviceA reduced the resource) reduces the same resource. Expected left amount of recources is 3, but we get 4.

Gotcha! I will fix it. Thanks.

> I am open to out-of-the-box suggestions for a nicer index (?) page that would make this feature more functional to its true use (which is not just to see a list of our available resources - that's just a small part of it), sadly I am not very good with front-end ideas.

I will add screenshot of few designs.

**Comment by differentreality** [comment]
@BelieveC are you going to fix how resources are being updated?

**Comment by BelieveC** [comment]
@differentreality, I have made the changes :+1:

**Comment by BelieveC** [comment]
@differentreality I have made the changes :+1:

File: app/controllers/admin/resources_controller.rb [hunk 8]
```diff
@@ -23,12 +23,25 @@ def create
...
-        render :edit
+        flag = @resource.update_attributes(resource_params)
+      end
+      respond_to do |format|
+        if flag
+          format.html do
+            redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
+                        notice: 'Resource successfully updated.'
+          end
+          flash[:notice] = "One #{params[:inc].to_i == 1 ? 'resource freed' : 'more resource used'}"
```
Comment (differentreality) [inline app/controllers/admin/resources_controller.rb]: For this flash message, you need to think about what  the organizer needs to know, and give the organizer the right information to make his/her life easier. 
I, for example, would care to know **which** resource changed, eg. did I just click on T-shirt size M or size L?

I don't think that this message is very informative or helpful.

File: app/controllers/admin/resources_controller.rb [hunk 9]
```diff
@@ -23,12 +23,25 @@ def create
...
+          format.js{ render action: 'update_quantity' }
+        else
+          flash[:error] = "Resource update failed: #{@resource.errors.full_messages.join('. ')}."
```
Comment (differentreality) [inline app/controllers/admin/resources_controller.rb]: Again, I need to know which resource failed to update, otherwise this offers the organizer very little meaningful information.

File: app/controllers/admin/resources_controller.rb [hunk 10]
```diff
@@ -23,12 +23,25 @@ def create
     end
 
     def update
-      if @resource.update_attributes(resource_params)
-        redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
-                    notice: 'Resource successfully updated.'
+      if params[:inc].present?
```
Comment (differentreality) [inline app/controllers/admin/resources_controller.rb]: What does inc mean? Please rename it to something that I can understand when reading the code.

**Comment by differentreality** [comment]
It is very confusing that I am clicking the + icon, but it does not mean that I just gave away 1 resource. The looks of the layout needs to be re-thought from an organizer's perspective. What does + mean for the organizer? When is the organizer going to use this interface and to do what?

File: app/controllers/admin/resources_controller.rb (hunk 10 above)
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Inc means increment. Alright, I will rename it.

File: app/controllers/admin/resources_controller.rb (hunk 9 above)
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Alright :+1:

File: app/controllers/admin/resources_controller.rb (hunk 8 above)
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: I will make the changes.

**Comment by BelieveC** [comment]
>What does + mean for the organizer? When is the organizer going to use this interface and to do what?

We can add tool pit for to give more information about what action would be performed when +/- is clicked. 

> The looks of the layout needs to be re-thought from an organizer's perspective

I agree with that. I'm trying to think about layout which would best suit for resource index. But nothing substantial came out. Will try for some more design. :+1:

**Comment by BelieveC** [comment]
@differentreality 

Toolpits of increment and decrement actions:

- Decrease used resource button
increment

- Increase used resource button
decrement

**Comment by differentreality** [comment]
We need to consider what + means for the organizer and provide an interface that does that is expected, not just add tooltips and messages. We need to make this easy for use. 
So what's our workflow here?
Organizer gives out an item -> clicks -  for deducting 1 thing (from the pile of items) that was given away (or sold in other cases), so in our words we decrease the remaining quantity. 

When implementing features like this you need to come up with a workflow first, so that we know what we want to do. 

You need to consider if the user (organizer) that uses OSEM will know what to do when presented with this page and sees - 23/50 + 
Is it obvious what needs to be done? Not immediately. What if we see the column's name, it says left( left/total). What it actually means is remaining and when we show the information we don't do
23( 23/50) as indicated on the column header, so perhaps the column header needs to be more accurate:
`Remaining / Total `

**Comment by BelieveC** [comment]
> When implementing features like this you need to come up with a workflow first, so that we know what we want to do.

Thanks! I will that in mind :smile:

**Comment by BelieveC** [comment]
@differentreality I don't think we make much change in the index page of resources with the constraint of displaying many resources on a single page. But How about adding Resource Image to resources. Following are benefits of Resource Image:

- Differentiate between two similar resources. e.g Chair1 and Chair2
- Easy to identify resource without reading the description and all.

What do you think?

File: app/views/admin/resources/update_quantity_error.js.erb [hunk 11]
```diff
@@ -0,0 +1 @@
+$('#messages').html("<%= j render 'layouts/messages' %>");
```
Comment (differentreality) [inline app/views/admin/resources/update_quantity_error.js.erb]: why not use unobtrusive flash? Then you shouldn't need this.
https://github.com/leonid-shevtsov/unobtrusive_flash

File: app/views/admin/resources/_update_actions.html.haml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+.btn-group
+  =link_to admin_conference_resource_path(conference.short_title, resource, used_increment_flag: 0), method: :put, title: "Use 1 #{resource.name}", class: "btn btn-primary action-btn #{ 'disabled' if resource.used == resource.quantity }", remote: true do
```
Comment (differentreality) [inline app/views/admin/resources/_update_actions.html.haml]: space after =

File: app/controllers/admin/resources_controller.rb [hunk 13]
```diff
@@ -23,12 +23,29 @@ def create
     end
 
     def update
-      if @resource.update_attributes(resource_params)
-        redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
-                    notice: 'Resource successfully updated.'
+      if params[:used_increment_flag].present?
```
Comment (differentreality) [inline app/controllers/admin/resources_controller.rb]: We need to work a lot on this controller, it's impossible to read.

File: app/views/admin/resources/update_quantity_error.js.erb (hunk 11 above)
Comment (BelieveC) [inline app/views/admin/resources/update_quantity_error.js.erb]: Yeah! I will use it :+1:

File: app/controllers/admin/resources_controller.rb (hunk 13 above)
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: I have refactored it now :+1:

File: app/views/admin/resources/_update_actions.html.haml (hunk 12 above)
Comment (BelieveC) [inline app/views/admin/resources/_update_actions.html.haml]: :+1:

**Comment by BelieveC** [comment]
@differentreality I have made the changes!

**Comment by differentreality** [comment]
I would try something like this for the index page, so that it responds better to screen size changes

screenshot from 2017-05-06 12-00-26
http://paste.opensuse.org/64030400

**Comment by differentreality** [comment]
The flash messages don't disappear.

File: app/controllers/admin/resources_controller.rb [hunk 14]
```diff
@@ -48,5 +56,22 @@ def destroy
...
+      else
+        @resource.update_attributes(resource_params)
+      end
+    end
+
+    def get_successful_update_flash(used_increment_flag)
+      if used_increment_flag.to_i == 1
+        "One #{@resource.name} freed."
+      else
+        "One more #{@resource.name} used."
```
Comment (differentreality) [inline app/controllers/admin/resources_controller.rb]: How/When do I access this case?

Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Update to the resource can be via Edit page or Increment/Decrement via Index page. `get_update_flag` action distinguish between above two updates in order to use proper update logic for a different scenario. `get_successful_update_flash` action is just used to generate flash for later update case. The sole purpose of the separate action was to make the code more readable.

**Comment by BelieveC** [comment]
@differentreality Check whether you find controller readable? or else I will try something else!

**Comment by BelieveC** [comment]
Fixed conflict in resources.rb

File: app/controllers/admin/resources_controller.rb [hunk 15]
```diff
@@ -48,5 +56,24 @@ def destroy
...
+        # Increment/Decrement of Resource via Index Page
+        @resource.used = params[:increment_used_resource_flag].to_i == 1 ? (@resource.used + 1) : (@resource.used - 1)
+        @resource.save
+      else
+        # Update via Edit Page
+        @resource.update_attributes(resource_params)
+      end
+    end
+
+    def get_successful_update_flash(increment_used_resource_flag)
```
Comment (differentreality) [inline app/controllers/admin/resources_controller.rb]: This function is not needed

File: app/controllers/admin/resources_controller.rb [hunk 16]
```diff
@@ -48,5 +56,24 @@ def destroy
     def resource_params
       params.require(:resource).permit(:name, :description, :quantity, :used, :conference_id)
     end
+
+    def check_successful_update(increment_used_resource_flag)
```
Comment (differentreality) [inline app/controllers/admin/resources_controller.rb]: This is not needed if you just have params[:used_change] which has values of -1 or +1

File: app/views/admin/resources/_update_links.html.haml [hunk 17]
```diff
@@ -0,0 +1,7 @@
+= link_to admin_conference_resource_path(conference.short_title, resource, increment_used_resource_flag: 1), method: :put, title: "Use 1 #{resource.name}", class: "btn btn-primary action-btn #{ 'disabled' if resource.used== resource.quantity }", remote: true do
```
Comment (differentreality) [inline app/views/admin/resources/_update_links.html.haml]: I think it makes more sense that I give out something (which is my default action as an info desk volunteer) and I click +
(which should increase `used`)

File: app/assets/javascripts/application.js [hunk 18]
```diff
@@ -57,3 +57,5 @@ $(document).ready(function() {
         delegateSelector: 'a.smoothscroll'
     });
 });
+
+UnobtrusiveFlash.flashOptions['timeout'] = 2000;
```
Comment (BelieveC) [inline app/assets/javascripts/application.js]: > That's 2 seconds, is that really enough time for someone to read a comment? 

It would vary person to person. Flash message have at max 3 words except for resource name. Apart from this, the reason I felt to keep it 2 second was just to assure the least number of flash message to appear on the page at any instant. More the number of messages, more resource content table would move towards bottom of page.

File: app/controllers/admin/resources_controller.rb [hunk 19]
```diff
@@ -23,12 +24,19 @@ def create
...
-        flash.now[:error] = "Resource update failed: #{@resource.errors.full_messages.join('. ')}."
-        render :edit
+      successful_update = check_successful_update params[:increment_used_resource_flag]
+      respond_to do |format|
+        if successful_update
+          format.html do
+            redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
+                        notice: 'Resource successfully updated.'
+          end
+          flash.now[:notice] = get_successful_update_flash params[:increment_used_resource_flag]
```
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: >Why do we need a function to define our flash message with a simple if statement?

I thought modularizing update action into smaller functions would make it more readable. I'm trying my best to make it readable by implementing different things. Will remove it.

File: app/views/admin/resources/index.html.haml [hunk 20]
```diff
@@ -20,17 +22,16 @@
...
+                  %span{ id: "resource_used_#{resource.id}" }
+                    = resource.used
                 %td
                   .btn-group
                     = link_to 'Edit', edit_admin_conference_resource_path(@conference.short_title, resource.id),
                       method: :get, class: 'btn btn-primary', disabled: !(can? :update, resource)
                     = link_to 'Delete', admin_conference_resource_path(@conference.short_title, resource.id),
                       method: :delete, class: 'btn btn-danger', disabled: !(can? :destroy, resource),
                       data: { confirm: "Do you really want to delete #{resource.name}?" }
-.row
```
Comment (BelieveC) [inline app/views/admin/resources/index.html.haml]: I asked for your opinion on this in the comment [link](https://github.com/openSUSE/osem/pull/1301#issuecomment-292151739). I will repeat what I proposed: 

- Unable to find it as there are paging 's next and previous action which occupies large space
- It is convenient to have important actions at the top rather than bottom e.g When we use google keep we have input to add a task at the top instead of the bottom.

Comment (BelieveC) [inline app/views/admin/resources/index.html.haml]: > Doesn't the whole application have this in this position?

Just a proposal. I will revert it if don't find it appropriate.

File: app/views/admin/resources/_update_links.html.haml (hunk 17 above)
Comment (BelieveC) [inline app/views/admin/resources/_update_links.html.haml]: Agree! i.e Our increment/decrement action should be shifted to `used` column right?

File: app/views/admin/resources/_update_actions.html.haml [hunk 21]
```diff
@@ -0,0 +1,5 @@
+.btn-group.hidden-sm.hidden-xs
```
Comment (BelieveC) [inline app/views/admin/resources/_update_actions.html.haml]: It can be! Will move it to index file :+1:

File: app/views/admin/resources/index.html.haml (hunk 20 above)
Comment (differentreality) [inline app/views/admin/resources/index.html.haml]: No, it is **not** appropriate if the 'New' button is sometimes on top, and sometimes on the bottom. This is not good for our design, not helpful to our users. You need to put where users can find it, based on their experience from other parts of the application. 
If you disagree with its position, that's a different issue you can bring up, and if decided so, it would change for the whole application, not just 1 page. Makes sense?

File: app/controllers/admin/resources_controller.rb (hunk 19 above)
Comment (differentreality) [inline app/controllers/admin/resources_controller.rb]: Try coming back to your code 1-2 days after you have submitted your PR. Then go to the changed files and read your changes. Does it help you to jump from place to place to find what the controller does?

Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Yeah! You are right! I think it will grow on me with time. :smile:

File: app/views/admin/resources/index.html.haml (hunk 20 above)
Comment (BelieveC) [inline app/views/admin/resources/index.html.haml]: Alright! I would open a separate issue regarding above proposal +1:

File: app/assets/stylesheets/osem.css.scss [hunk 22]
```diff
@@ -89,4 +89,15 @@ p.comment-body {
...
-}
\ No newline at end of file
+}
+
+.quantity_left{
+  height: 3em;
+  width: 9em;
+}
+
+@media (max-width: 767px){
```
Comment (BelieveC) [inline app/assets/stylesheets/osem.css.scss]: Previously required to make buttoms responsive but as we have added `.visible-sm.visible-xs` class It is no longer required! Removed it.

File: app/controllers/admin/resources_controller.rb (hunk 15 above)
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Is [this](https://github.com/openSUSE/osem/pull/1301/commits/29eedd92cc5d2484d551a2625cd797961cff8664) what you want instead of function??

File: app/controllers/admin/resources_controller.rb (hunk 16 above)
Comment (BelieveC) [inline app/controllers/admin/resources_controller.rb]: Are you suggesting to carry out update this way 
`@resource.update(name: params[:resource][:name],
                            description: params[:resource][:description],
                            used: params[:resource][:used].to_i + params[:used_count].to_i,
                            quantity: params[:resource][:quantity]
                           )`
Or something else?

**Comment by BelieveC** [comment]
@differentreality, Can you review it? Let's finish it off within this month.

**Comment by differentreality** [comment]
@BelieveC please squash your commits and rebase!

**Comment by BelieveC** [comment]
@differentreality, Sorry It took long as I couldn't test it as environment was failing to build.

Rebased :+1:

**Comment by differentreality** [comment]
@BelieveC nice, let's work on ideas that improve the UI.

**Comment by BelieveC** [comment]
@differentreality Currently It looks like this

screenshot from 2017-08-09 14-41-31

I find it clean and simple. Any suggestion about design?

**Comment by differentreality** [comment]
@BelieveC the issue was about refactoring how this looks. If you don't have any ideas for a significant change that can be beneficial to organizers, perhaps you should close this PR, until there are more ideas on the table, which we can discuss on the issue.

**Comment by BelieveC** [comment]
@differentreality, Alright!
